invenio_circulation/errors.py

```diff
@@ -8,50 +8,142 @@
+
 
 class NoValidTransitionAvailable(CirculationException):
     """Exception raised when all transitions conditions failed."""
 
+    def __init__(self, loan_pid=None, state=None, **kwargs):
+        """Initialize exception."""
+        super(NoValidTransitionAvailable, self).__init__(**kwargs)
+        self.description = (
+            "For the loan with pid '{0}' there are no valid transitions from "
+            "its current state '{1}'".format(loan_pid, state)
+        )
 
-class LoanActionError(CirculationException):
-    """Exception raised when invalid action triggered on a loan."""
 
+# Loans
+class InvalidLoanState(CirculationException):
+    """State not found in circulation configuration."""
 
-class InvalidCirculationPermission(CirculationException):
-    """Raised when permissions are not satisfied for transition."""
+    def __init__(self, state=None, **kwargs):
+        """Initialize exception."""
+        self.description = "Invalid loan state '{}'".format(state)
+        super(InvalidLoanState, self).__init__(**kwargs)
 
 
-class TransitionConstraintsViolation(CirculationException):
-    """Exception raised when constraints for the transition failed."""
+class ItemNotAvailable(CirculationException):
+    """Exception raised from action on unavailable item."""
 
+    def __init__(self, item_pid=None, transition=None, **kwargs):
+        """Initialize exception."""
+        super(ItemNotAvailable, self).__init__(**kwargs)
+        self.description = (
+            "The item requested with pid '{0}' is not available. "
+            "Transition to '{1}' has failed.".format(item_pid, transition)
+        )
 
-class MultipleLoansOnItemError(CirculationException):
-    """Exception raised when more than one loan on an item."""
+
+class ItemDoNotMatch(CirculationException):
+    """Exception raised from action on unavailable item."""
 
 
-class NotImplementedError(CirculationException):
+class MultipleLoansOnItem(CirculationException):
     """Exception raised when more than one loan on an item."""
+
+    def __init__(self, item_pid=None, **kwargs):
+        """Initialize exception."""
+        super(MultipleLoansOnItem, self).__init__(**kwargs)
+        self.description = (
+            "Multiple active loans on item with pid '{}'".format(item_pid)
+        )
+
+
+class LoanMaxExtension(CirculationException):
+    """Exception raised when reached the max extensions for a loan."""
+
+    def __init__(self, loan_pid=None, extension_count=None, **kwargs):
+        """Initialize exception."""
+        super(LoanMaxExtension, self).__init__(**kwargs)
+        self.description = (
+            "You have reached the maximum amount of extesions '{}' "
+            "for loan '{}'".format(extension_count, loan_pid)
+        )
+
+
+# General
+class RecordCannotBeRequested(CirculationException):
+    """Exception raised when item can not be requested."""
+
+
+class NotImplementedCirculation(CirculationException):
```

Comment:
I think the name is a bit misleading in comparison when this exception is raised, maybe as something like `ConfigurationVariableMissing`?